src/script/script.js

```diff
@@ -0,0 +1,402 @@
+  generateKeyboardRow(array) {
+    if (localStorage.keyboardLanguage !== "ru" && localStorage.keyboardLanguage !== "en") {
+      localStorage.keyboardLanguage = "ru";
+    }
+    let row = document.createElement('div');
+    row.classList.add('keyboard--row');
+    array.forEach(elem => {
+      row.append(this.generateKeyboardButton(elem));
+    });
+    return row;
+  }
+
+  generateKeyboardButton(obj) {
+    let btn = document.createElement('div');
+    btn.classList.add("keyboard--btn", obj.className);
+    if (obj.symbol) {
+      btn.classList.add("symbol");
+      let current = localStorage.keyboardLanguage === "ru" ? obj.ru.down : obj.en.down;
+      btn.dataset.value = current;
+    };
+
+    let output = !obj.symbol ? `<span>${obj.value}</span>` :
+      `<span class="ru ${localStorage.keyboardLanguage === "ru" ? "" : "hidden"}">
+          <span class="caseDown">${obj.ru.down}</span>
+          <span class="caseUp hidden">${obj.ru.up}</span>
+          <span class="caps hidden">${obj.ru.caps}</span>
+          <span class="shiftCaps hidden">${obj.ru.shiftcaps}</span>
+        </span>
+        <span class="en ${localStorage.keyboardLanguage === "en" ? "" : "hidden"}">
+          <span class="caseDown">${obj.en.down}</span>
+          <span class="caseUp hidden">${obj.en.up}</span>
+          <span class="caps hidden">${obj.en.caps}</span>
+          <span class="shiftCaps hidden">${obj.en.shiftcaps}</span>
+        </span>`;
+    btn.innerHTML = output;
+    return btn;
+  }
+
+  create() {
+    this.generateHeader();
+    this.generateTextArea();
+    this.generateKeyboard();
+  }
+
+  remove() {
+    this.rootElement.remove();
+  }
+
+  ///////////////////////////////////////////////////////////////
+  switchLanguage() {
+    localStorage.keyboardLanguage = localStorage.keyboardLanguage === "ru" ? "en" : "ru";
+    for (let key of this.keys) {
+      let items = key.children;
+      Array.from(items).forEach(el => {
+        if (el.classList.contains(localStorage.keyboardLanguage)) {
+          el.classList.remove('hidden');
+          this.toggleClassesInButton(key, "caseDown");
+        } else {
+          el.classList.add('hidden');
+        }
+      });
+    }
+  }
+
+  textAreaChangeInfo(value) {
+    let text = this.textArea.value;
+    this.textArea.focus();
+    if (value === "&amp;"){
+      this.textArea.setRangeText("&", this.textArea.selectionStart, this.textArea.selectionEnd, "end");
+    } else if (value === "Backspace") {
+      if (this.currentPosition) {
+        this.textArea.value = text.slice(0, this.currentPosition - 1) + '' + text.slice(this.currentPosition, text.length);
+        this.textArea.setRangeText("", this.currentPosition, this.currentPosition, "end");
+      } else {
+        this.textArea.value = text.slice(0, -1);
+      }
+    } else if (value === "Delete"){
+      this.textArea.value = text.slice(0, this.currentPosition) + '' + text.slice(this.currentPosition+1, text.length);
```

Comment:
DRY.
You use almost the same code for different cases. Try to highlight the common part

Reply:
Replaced to switch case.